packages/share/android/src/main/java/io/flutter/plugins/share/SharePlugin.java

```diff
@@ -4,59 +4,63 @@
+public class SharePlugin implements FlutterPlugin, ActivityAware {
 
   private static final String CHANNEL = "plugins.flutter.io/share";
+  private MethodCallHandler handler;
+  private Activity activity;
+  private Share share;
+  private MethodChannel methodChannel;
 
   public static void registerWith(Registrar registrar) {
-    MethodChannel channel = new MethodChannel(registrar.messenger(), CHANNEL);
-    SharePlugin instance = new SharePlugin(registrar);
-    channel.setMethodCallHandler(instance);
+    SharePlugin plugin = new SharePlugin();
+    plugin.setUpChannel(registrar.messenger());
   }
 
-  private final Registrar mRegistrar;
+  @Override
+  public void onAttachedToEngine(FlutterPluginBinding binding) {
+    setUpChannel(binding.getFlutterEngine().getDartExecutor());
+  }
+
+  @Override
+  public void onDetachedFromEngine(FlutterPluginBinding binding) {}
 
-  private SharePlugin(Registrar registrar) {
-    this.mRegistrar = registrar;
+  @Override
+  public void onAttachedToActivity(ActivityPluginBinding binding) {
+    activity = binding.getActivity();
+    share.setActivity(activity);
+  }
+
+  @Override
+  public void onDetachedFromActivity() {
+    activity = null;
+    share.setActivity(null);
+    methodChannel.setMethodCallHandler(null);
+  }
+
+  @Override
+  public void onReattachedToActivityForConfigChanges(ActivityPluginBinding binding) {
+    onAttachedToActivity(binding);
   }
 
   @Override
-  public void onMethodCall(MethodCall call, MethodChannel.Result result) {
-    if (call.method.equals("share")) {
-      if (!(call.arguments instanceof Map)) {
-        throw new IllegalArgumentException("Map argument expected");
-      }
-      // Android does not support showing the share sheet at a particular point on screen.
-      share((String) call.argument("text"), (String) call.argument("subject"));
-      result.success(null);
-    } else {
-      result.notImplemented();
-    }
+  public void onDetachedFromActivityForConfigChanges() {
+    onDetachedFromActivity();
   }
 
-  private void share(String text, String subject) {
-    if (text == null || text.isEmpty()) {
-      throw new IllegalArgumentException("Non-empty text expected");
-    }
-
-    Intent shareIntent = new Intent();
-    shareIntent.setAction(Intent.ACTION_SEND);
-    shareIntent.putExtra(Intent.EXTRA_TEXT, text);
-    shareIntent.putExtra(Intent.EXTRA_SUBJECT, subject);
-    shareIntent.setType("text/plain");
-    Intent chooserIntent = Intent.createChooser(shareIntent, null /* dialog title optional */);
-    if (mRegistrar.activity() != null) {
-      mRegistrar.activity().startActivity(chooserIntent);
-    } else {
-      chooserIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-      mRegistrar.context().startActivity(chooserIntent);
-    }
+  private void setUpChannel(BinaryMessenger messenger) {
+    methodChannel = new MethodChannel(messenger, CHANNEL);
+    share = new Share(activity);
```

Comment:
It looks like `activity` is always null at this point. You can probably remove the local `activity` variable and exclusively use the setter method, instead.